queue.c

```diff
@@ -2730,24 +2730,37 @@ BaseType_t xQueueIsQueueFullFromISR( const QueueHandle_t xQueue )
         configASSERT( xQueue );
         configASSERT( pcQueueName );
 
+        QueueRegistryItem_t * pxEntryToWrite = NULL;
+
         /* See if there is an empty space in the registry.  A NULL name denotes
          * a free slot. */
         for( ux = ( UBaseType_t ) 0U; ux < ( UBaseType_t ) configQUEUE_REGISTRY_SIZE; ux++ )
         {
-            if( xQueueRegistry[ ux ].pcQueueName == NULL )
+            /* Replace an existing entry if the queue is already in the registry. */
+            if( xQueueRegistry[ ux ].xHandle == xQueue )
             {
-                /* Store the information on this queue. */
-                xQueueRegistry[ ux ].pcQueueName = pcQueueName;
-                xQueueRegistry[ ux ].xHandle = xQueue;
-
-                traceQUEUE_REGISTRY_ADD( xQueue, pcQueueName );
+                pxEntryToWrite = &( xQueueRegistry[ ux ] );

```

Comment:
If the queue handle is already in the registry, the code breaks out of the loop and overwrites the name asssociated with that queue handle.